pkg/operator/status.go

```diff
@@ -69,19 +76,56 @@ func isDeploymentStatusComplete(deploy *appsapi.Deployment) bool {
 		deploy.Status.ObservedGeneration >= deploy.Generation
 }
 
+func (c *Controller) setStatusRemoving(cr *imageregistryv1.Config) {
+	operatorProgressing := clusteroperator.ConditionState{
+		Status:  configapiv1.ConditionTrue,
+		Message: "Registry is being removed",
+		Reason:  "Removing",
+	}
+
+	err := c.clusterStatus.Update(configapiv1.OperatorProgressing, operatorProgressing, "")
+	if err != nil {
+		glog.Errorf("unable to update cluster status to %s=%s: %s", configapiv1.OperatorProgressing, configapiv1.ConditionTrue, err)
+	}
+
+	updateCondition(cr, operatorapiv1.OperatorStatusTypeProgressing, operatorProgressing)
+}
+
+func (c *Controller) setStatusRemoveFailed(cr *imageregistryv1.Config) {
+	operatorFailing := clusteroperator.ConditionState{
+		Status:  configapiv1.ConditionTrue,
+		Message: "Unable to remove registry",
+		Reason:  "RemoveFailed",
+	}
+
+	err := c.clusterStatus.Update(configapiv1.OperatorFailing, operatorFailing, "")
+	if err != nil {
+		glog.Errorf("unable to update cluster status to %s=%s: %s", configapiv1.OperatorFailing, configapiv1.ConditionTrue, err)
+	}
+
+	updateCondition(cr, operatorapiv1.OperatorStatusTypeFailing, operatorFailing)
+}
+
 func (c *Controller) syncStatus(cr *imageregistryv1.Config, deploy *appsapi.Deployment, applyError error, removed bool) {
 	setOperatorVersion := false
-	operatorAvailable := osapi.ConditionFalse
-	operatorAvailableMsg := ""
+	operatorAvailable := clusteroperator.ConditionState{
+		Status:  configapiv1.ConditionFalse,
+		Message: "",
+		Reason:  "",
+	}
 	if deploy == nil {
-		operatorAvailableMsg = "Deployment does not exist"
+		operatorAvailable.Message = "Deployment does not exist"
+		operatorAvailable.Reason = "DeploymentNotFound"
 	} else if deploy.DeletionTimestamp != nil {
-		operatorAvailableMsg = "Deployment is being deleted"
+		operatorAvailable.Message = "Deployment is being deleted"
+		operatorAvailable.Reason = "DeploymentDeleted"
 	} else if !isDeploymentStatusAvailable(deploy) {
-		operatorAvailableMsg = "Deployment does not have available replicas"
+		operatorAvailable.Message = "Deployment does not have available replicas"
+		operatorAvailable.Reason = "NoReplicasAvailable"
 	} else {
-		operatorAvailable = osapi.ConditionTrue
-		operatorAvailableMsg = "Deployment has minimum availability"
+		operatorAvailable.Status = configapiv1.ConditionTrue
+		operatorAvailable.Message = "Deployment has minimum availability"
```

Comment:
can we distinguish between "minimal availability" and "we're fully deployed" and convey that to the user?  As an admin i'd be nervous seeing "deployment has minimum availability" because it sounds like a warning.  I'd be wondering why it's only at the minimum.